tests/conftest.py

```diff
@@ -39,6 +39,9 @@ def pytest_addoption(parser):
     os.environ['PIP_DISABLE_PIP_VERSION_CHECK'] = '1'  # do not pollute stderr with upgrade advisory
     parser.addoption('--run-integration', action='store_true', help='run the integration tests')
     parser.addoption('--only-integration', action='store_true', help='only run the integration tests')
+    if platform.python_implementation() == 'PyPy' and sys.version_info[0] == 2 and os.name == 'nt':
+        # pytest-xdist does not work on Windows PyPy2, disable it => last compatible: pytest 4 + pytest-xdist 1.31
+        next(o for o in parser.getgroup('xdist').options if o.dest == 'numprocesses')._attrs['type'] = lambda s: None  # noqa
```

Comment:
There's a lot going on in this one-liner. I recommend breaking it up into smaller chunks.

numprocesses = next(o for o in parser.getgroup('xdist').options if o.dest == 'numprocesses')
numprocesses._attrs['type'] = lambda s: None